src/main/java/ch/njol/skript/expressions/ExprCraftingSlots.java

```diff
@@ -0,0 +1,167 @@
+	})
+@Since("INSERT VERSION")
+public class ExprCraftingSlots extends SimpleExpression<ItemStack> {
+	
+	static {
+		Skript.registerExpression(ExprCraftingSlots.class, ItemStack.class, ExpressionType.COMBINED,
+				"[the] crafting [inventory] result (slot|item) [of %inventories%]",
+				"%inventories%'[s] crafting [inventory] result (slot|item)",
+				"[the] crafting [inventory] (matrix|grid) [(slots|items|shape)] [of %inventories%]",
+				"%inventories%'[s] crafting [inventory] (matrix|grid) [(slots|items|shape)]");
+	}
+
+	@SuppressWarnings("NotNullFieldNotInitialized")
+	private Expression<Inventory> invis;
+	private boolean isMatrix;
+	private final ItemStack AIR = new ItemStack(Material.AIR);
+	private final ItemStack[] AIR_ITEMSTACK_LIST = new ItemStack[]{AIR, AIR, AIR, AIR, AIR, AIR, AIR, AIR, AIR};
+
+	@Override
+	@SuppressWarnings({"null", "unchecked"})
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
+		invis = (Expression<Inventory>) exprs[0];
+		isMatrix = matchedPattern > 1;
+		return true;
+	}
+
+	@Override
+	@Nullable
+	protected ItemStack[] get(Event e) {
+		Inventory[] invis = this.invis.getArray(e);
+		if (invis == null)
+			return null;
+
+		List<ItemStack> items = new ArrayList<>();
+		for (Inventory invi : invis) {
+			if (!(invi instanceof CraftingInventory))
+				continue;
+
+			CraftingInventory craft = (CraftingInventory) invi;
+			if (isMatrix) {
+				items.addAll(Arrays.asList(craft.getMatrix()));
+			} else {
+				items.add(craft.getResult());
+			}
+		}
+		return items.toArray(new ItemStack[0]);
+	}
+
+	@Override
+	@Nullable
+	public Class<?>[] acceptChange(ChangeMode mode) {
+		if (mode == ChangeMode.SET || mode == ChangeMode.DELETE)
+			return CollectionUtils.array(ItemStack[].class);
+
+		return null;
+	}
+
+	@Override
+	public void change(Event e, @Nullable Object[] delta, ChangeMode mode) {
+		Inventory[] invis = this.invis.getArray(e);
+		if (invis == null)
+			return;
+
+		if (e instanceof PrepareItemCraftEvent && isMatrix) { // Until spigot/paper fixes this
+			Skript.warning("It is not currently possible to edit the crafting inventory matrix in a preparing craft event.");
+			return;
+		}
+
+		if (mode == ChangeMode.DELETE) {
+			for (Inventory invi : invis) {
+				if (!(invi instanceof CraftingInventory))
+					continue;
+				if (isMatrix)
+					((CraftingInventory) invi).setMatrix(AIR_ITEMSTACK_LIST);
+				else
+					((CraftingInventory) invi).setResult(new ItemStack(Material.AIR));
+			}
+		} else { // SET
+			assert invis != null;
+			ItemStack[] items = (ItemStack[]) delta;
+			if (items == null)
+				return;
+			for (Inventory invi : invis) {
+				if (!(invi instanceof CraftingInventory))
+					continue;
+				if (isMatrix) {
+					ItemStack[] itemStacks = new ItemStack[9];
+					for (int i = 0; i < 9; i++) { // list must be 9 if not we will fill it manually with AIR
+						itemStacks[i] = i < items.length ? items[i] : AIR;
```

Comment:
It's not an issue to use the same itemstack multiple times right?

Reply:
Shouldn't be